File: day1/part_2_solution.rb
def read_file(file_name)
  file = File.open(file_name, "r")
  data = file.read
  file.close
  return data
end

file_name = "input.txt"
arr = read_file(file_name).split("\n").map(&:to_i)
total = arr[0] + arr[1] + arr[2]
last = total
sum = 0
for i in 3...arr.size do
  total -= arr[i-3]
  total += arr[i]
  sum += 1 if total > last
  last = total
end
puts sum

File: 2022/day6/part_2_solution.js
const fs = require('fs');

const readFile = (path) => {
  return fs.readFileSync(path, { encoding: 'utf8' });
}

const solution = (input) => {
  let arr = input.split("");
  for(let i=13; i<arr.length; i++) {
    let s = new Set();
    for (let j=i; j>=i-13; j--) {
      s.add(arr[j]);
    }
    if (s.size === 14) {
      console.log(i+1);
      break;
    }
  }
}

let fileName = "example";

if (process.argv[2]) {
  const maybe = process.argv[2].split("--input=");
  if(maybe.length === 2) {
    fileName = maybe[1];
  }
}

const fileContent = readFile(`./${fileName}.txt`);
solution(fileContent);